.github/workflows/rust.yml

```diff
@@ -32,6 +32,22 @@ jobs:
     - name: Check style
       run: cargo fmt -- --check
 
+  check-omicron-deployment:
```

Comment:
I wonder if this will be a lot slower (or faster?) than making this a step in the `build-and-test` job.  As a step in the `build-and-test` job, we may not need to rebuild as many things.  But maybe the point is to rebuild them, and this will be run by GitHub in parallel, so maybe it doesn't matter?